src/single-line-diagram-viewer.ts

```diff
@@ -696,38 +699,41 @@ export class SingleLineDiagramViewer {
         const equipmentsWithPopover = ['LINE', 'TWO_WINDINGS_TRANSFORMER'];
 
         // handling the hover on the equipments
-        const svgEquipments = this.svgMetadata.nodes.filter((node) =>
+        const svgEquipments = this.svgMetadata?.nodes.filter((node) =>
             equipmentsWithPopover.includes(node.componentType)
         );
-        svgEquipments.forEach((equipment) => {
+        svgEquipments?.forEach((equipment) => {
             const svgEquipment = this.container?.querySelector(
                 '#' + equipment.id
             );
-
-            svgEquipment.addEventListener('mouseover', (event) => {
-                this.handleTogglePopover(
-                    true,
-                    event.currentTarget,
-                    equipment.equipmentId,
-                    equipment.componentType
-                );
+            svgEquipment?.addEventListener('mouseover', (event) => {
+                this.handleTogglePopover &&
+                    this.handleTogglePopover(
+                        true,
+                        event.currentTarget,
+                        equipment.equipmentId,
+                        equipment.componentType
+                    );
             });
-            svgEquipment.addEventListener('mouseout', () => {
-                this.handleTogglePopover(false, null, '', '');
+            svgEquipment?.addEventListener('mouseout', () => {
+                this.handleTogglePopover &&
+                    this.handleTogglePopover(false, null, '', '');
             });
         });
     }
 
     private addBusHandler() {
-        const buses = this.svgMetadata.nodes.filter((element) =>
+        const buses = this.svgMetadata?.nodes.filter((element) =>
             BUSBAR_SECTION_TYPES.has(element.componentType)
         );
-        buses.forEach((bus) => {
-            const svgBus: HTMLElement | null | undefined =
-                this.container?.querySelector('#' + bus.id);
-            svgBus.style.cursor = 'pointer';
-
-            svgBus.addEventListener('contextmenu', (event) => {
+        buses?.forEach((bus) => {
+            const svgBus: HTMLElement | null = this.container?.querySelector(
+                '#' + bus.id
+            );
+            if (svgBus) {
+                svgBus.style.cursor = 'pointer';
+            }
+            svgBus?.addEventListener('contextmenu', (event) => {
```

Comment:
Maybe move svgBus.addEventListener in the if above and you can remove the '?'

Reply:
Indeed. It's done